packages/patterns/test/test-patterns.js

```diff
@@ -1,27 +1,70 @@
+/* eslint-disable no-continue */
 import { test } from './prepare-test-env-ava.js';
 // eslint-disable-next-line import/order
 import { makeTagged } from '@endo/marshal';
 import { makeCopyBag, makeCopyMap, makeCopySet } from '../src/keys/checkKey.js';
 import { mustMatch, matches, M } from '../src/patterns/patternMatchers.js';
 import '../src/types.js';
 
+const { Fail } = assert;
+
 const runTests = (successCase, failCase) => {
+  /**
+   * @callback makeErrorMessage
+   * @param {string} repr
+   * @param {string} [kind]
+   * @param {string} [type]
+   * @returns {string}
+   */
+  /**
+   * Methods corresponding with pattern matchers that don't look past type.
+   *
+   * @type {Record<methodName: string, makeErrorMessage: makeErrorMessage}
+   */
+  const simpleMethods = {
+    any: _repr => Fail`must not expect rejection by M.any()`,
+    and: _repr => Fail`must not expect rejection by M.and()`,
+
+    scalar: (repr, kind, type = kind) =>
+      `A "${type}" cannot be a scalar key: ${repr}`,
+    key: (repr, kind) => `A passable tagged "${kind}" is not a key: ${repr}`,
+    pattern: _repr => Fail`M.pattern() rejection messages must be customized`,
+
+    boolean: (repr, kind) => `${kind} ${repr} - Must be a boolean`,
+    number: (repr, kind) => `${kind} ${repr} - Must be a number`,
+    bigint: (repr, kind) => `${kind} ${repr} - Must be a bigint`,
+    string: (repr, kind) => `${kind} ${repr} - Must be a string`,
+    symbol: (repr, kind) => `${kind} ${repr} - Must be a symbol`,
+    record: (repr, kind) => `${kind} ${repr} - Must be a copyRecord`,
+    array: (repr, kind) => `${kind} ${repr} - Must be a copyArray`,
+    set: (repr, kind) => `${kind} ${repr} - Must be a copySet`,
+    bag: (repr, kind) => `${kind} ${repr} - Must be a copyBag`,
+    map: (repr, kind) => `${kind} ${repr} - Must be a copyMap`,
+    remotable: (repr, kind) => `${kind} ${repr} - Must be a remotable`,
+    error: (repr, kind) => `${kind} ${repr} - Must be a error`,
+    promise: (repr, kind) => `${kind} ${repr} - Must be a promise`,
+    undefined: (repr, kind) => `${kind} ${repr} - Must be a undefined`,
+    null: repr => `${repr} - Must be: null`,
```

Comment:
It's interesting to note that `M.undefined()` is implemented as `makeKindMatcher('undefined')` (and therefore `makeTagged('match:kind', 'undefined')`) while `M.null()` is implemented as just `null`, resulting in the need for special-casing negative tests of the latter.

Reply:
That seems like an adequate list of irregularities that you should probably switch `M.null()` to be more like `M.undefined()`. This PR seems like a fine time to do that.

Reply:
> because it results in exceptions rather than false results when checking against a non-Key specimen such as a Matcher

The fact that this case acts differently itself seems like a bug. Agree?